eng/pipelines/templates/stages/source-build-stages.yml

```diff
@@ -0,0 +1,242 @@
+          runTests: false
+          extraProperties: "/p:DisableDevBuildAsDefaultForSourceOnly=true"
+        buildFromArchive: ${{ leg.buildFromArchive }}
+        runOnline: ${{ leg.runOnline }}
+        useMonoRuntime: ${{ leg.useMonoRuntime }}
+        withPreviousSDK: ${{ leg.withPreviousSDK }}
+
+- ${{ if eq(parameters.useMicrosoftBuildAssetsForTests, true) }}:
+  - stage: VMR_SourceOnly_Validation
+    displayName: VMR Source-Only Validation
+    dependsOn:
+    - VMR_Vertical_Build
+    - VMR_SourceOnly_Build
+    variables:
+    - template: ../variables/vmr-build.yml
+      parameters:
+        vmrBranch: ${{ variables.VmrBranch }}
+        buildSourceOnly: true
+    jobs:
+    - ${{ each leg in parameters.legs }}:
+      - template: ../jobs/vmr-build.yml
+        parameters:
+          # CentOS
+          ${{ if eq(leg.distro, 'centos') }}:
+            buildName: ${{ format(leg.buildNameFormat, variables.centOSStreamName) }}_Validation
+            container:
+              name: ${{ variables.centOSStreamContainerName }}
+              image: ${{ variables.centOSStreamContainerImage }}
+          # Fedora
+          ${{ elseif eq(leg.distro, 'fedora') }}:
+            buildName: ${{ format(leg.buildNameFormat, variables.fedoraName) }}_Validation
+            container:
+              name: ${{ variables.fedoraContainerName }}
+              image: ${{ variables.fedoraContainerImage }}
+          # Ubuntu
+          ${{ elseif eq(leg.distro, 'ubuntu') }}:
+            buildName: ${{ format(leg.buildNameFormat, variables.ubuntuName) }}_Validation
+            ${{ if eq(leg.targetArchitecture, 'arm64') }}:
+              container:
+                name: ${{ variables.ubuntuArmContainerName }}
+                image: ${{ variables.ubuntuArmContainerImage }}
+            ${{ else }}:
+              container:
+                name: ${{ variables.ubuntuContainerName }}
+                image: ${{ variables.ubuntuContainerImage }}
+          # AlmaLinux
+          ${{ elseif eq(leg.distro, 'almalinux') }}:
+            buildName: ${{ format(leg.buildNameFormat, variables.almaLinuxName) }}_Validation
+            container:
+              name: ${{ variables.almaLinuxContainerName }}
+              image: ${{ variables.almaLinuxContainerImage }}
+            targetRid: ${{ variables.almaLinuxX64Rid }}
+          # Alpine
+          ${{ elseif eq(leg.distro, 'alpine') }}:
+            buildName: ${{ format(leg.buildNameFormat, variables.alpineName) }}_Validation
+            container:
+              name: ${{ variables.alpineContainerName }}
+              image: ${{ variables.alpineContainerImage }}
+            targetRid: ${{ variables.alpineX64Rid }}
+
+          isBuiltFromVmr: ${{ parameters.isBuiltFromVmr }}
+          vmrBranch: ${{ variables.VmrBranch }}
+          targetArchitecture: ${{ leg.targetArchitecture }}
+          ${{ if eq(leg.targetArchitecture, 'arm64') }}:
```

Comment:
Thoughts on define a variable for the pool so that this if..else logic isn't duplicated between the build/validate stages?

Reply:
We can't define a variable in this scope. The only place we can define a variable here is at the stage level. But that's above the context of knowing what the architecture is because that's leg-specific. You could introduce another wrapper template around vmr-build.yml but then you just have that much more duplication of template parameters that you need to define to get the other inputs you need.